File: 백준/Silver/7568. 덩치/덩치.py
#7568 덩치
# 내 덩치는 아마 우주 꼴찌.. 이것도 for문을 이용한 브루트포스 문제

num = int(input())
dungchi = []
for i in range(num):
    weight, height = map(int, input().split(" "))
    dungchi.append((weight,height))

for i in dungchi:
    count = 1
    for j in dungchi:
        if i[0] <j[0] and i[1] < j[1]:
            count += 1

    print(count, end = " ")

File: 백준/Silver/7568. 덩치/덩치.kt
import java.io.*
fun main(){
    val br = BufferedReader(InputStreamReader(System.`in`))
    val bw = BufferedWriter(OutputStreamWriter(System.out))
    val num = br.readLine().toInt()
    val powerList = ArrayList<Pair<Int,Int>>()
    repeat(num){
        val input = br.readLine().split(" ")
        powerList.add(Pair(input[0].toInt(), input[1].toInt()))
    }
    for(i in 0 until num){
        var count = 1
        for(j in 0 until num){
            if(powerList[i].first<powerList[j].first && powerList[i].second<powerList[j].second){
                count++
            }
        }
        bw.write(count.toString())
        bw.write(" ")
    }
    bw.flush()
    bw.close()
    br.close()
}